dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -0,0 +1,250 @@
+package com.dat3m.dartagnan.encoding;
+
+import com.dat3m.dartagnan.program.analysis.BranchEquivalence;
+import com.dat3m.dartagnan.program.analysis.ExecutionAnalysis;
+import com.dat3m.dartagnan.program.analysis.alias.AliasAnalysis;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.core.MemEvent;
+import com.dat3m.dartagnan.program.event.core.utils.RegWriter;
+import com.dat3m.dartagnan.verification.Context;
+import com.dat3m.dartagnan.verification.VerificationTask;
+import com.dat3m.dartagnan.wmm.relation.Relation;
+import com.dat3m.dartagnan.wmm.utils.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+import org.sosy_lab.common.configuration.Option;
+import org.sosy_lab.common.configuration.Options;
+import org.sosy_lab.java_smt.api.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.dat3m.dartagnan.configuration.OptionNames.IDL_TO_SAT;
+import static com.dat3m.dartagnan.configuration.OptionNames.MERGE_CF_VARS;
+import static com.dat3m.dartagnan.program.event.Tag.INIT;
+import static com.dat3m.dartagnan.program.event.Tag.WRITE;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Options
+public final class EncodingContext {
+
+    private static final Logger logger = LogManager.getLogger(EncodingContext.class);
+
+    private final VerificationTask verificationTask;
+    private final Context analysisContext;
+    private final ExecutionAnalysis executionAnalysis;
+    private final AliasAnalysis aliasAnalysis;
+    private final SolverContext solverContext;
+    private final FormulaManager formulaManager;
+    private final BooleanFormulaManager booleanFormulaManager;
```

Comment:
Those two have multiple usages, in utility methods that are called thousands of times.  Since that class usually has one instance per runtime, the storage costs are almost non-existent.  We won't even save one import line by removing it.